tests/test_cli_compile.py

```diff
@@ -612,38 +630,31 @@ def test_local_url_package(
     assert dependency in out.stderr
 
 
-@pytest.mark.parametrize(
-    ("line", "dependency", "rewritten_line"),
-    (
-        pytest.param(
-            os.path.join(
-                MINIMAL_WHEELS_PATH, "small_fake_with_deps-0.1-py2.py3-none-any.whl"
-            ),
-            "\nfile:small_fake_with_deps-0.1-py2.py3-none-any.whl",
-            "file:small_fake_with_deps-0.1-py2.py3-none-any.whl",
-            id="Relative URL",
+def test_relative_file_uri_package(pip_conf, runner):
+    # Copy wheel into temp dir
+    shutil.copy(
+        os.path.join(
+            MINIMAL_WHEELS_PATH, "small_fake_with_deps-0.1-py2.py3-none-any.whl"
         ),
-        pytest.param(
-            os.path.join(
-                MINIMAL_WHEELS_PATH, "small_fake_with_deps-0.1-py2.py3-none-any.whl"
-            ),
-            "\nsmall-fake-with-deps"
-            " @ file://{absolute_path}/small_fake_with_deps-0.1-py2.py3-none-any.whl",
-            "small-fake-with-deps"
-            " @ file://{absolute_path}/small_fake_with_deps-0.1-py2.py3-none-any.whl",
-            id="Relative URL",
-        ),
-    ),
-)
-def test_relative_url_package(pip_conf, runner, line, dependency, rewritten_line):
-    dependency = dependency.format(absolute_path=Path(".").absolute())
-    rewritten_line = rewritten_line.format(absolute_path=Path(".").absolute())
-    shutil.copy(line, ".")
+        ".",
+    )
     with open("requirements.in", "w") as req_in:
-        req_in.write(dependency)
+        req_in.write("file:small_fake_with_deps-0.1-py2.py3-none-any.whl")
     out = runner.invoke(cli, ["-n", "--rebuild"])
     assert out.exit_code == 0
-    assert rewritten_line in out.stderr
+    assert "file:small_fake_with_deps-0.1-py2.py3-none-any.whl" in out.stderr
+
+
+def test_direct_reference_with_extras(runner):
```

Comment:
@atugushev new unit test to confirm extras with direct reference is working and no need to add any logic to the existing codebase.